nuxeo-drive-jsf/src/test/java/org/nuxeo/drive/seam/TestNuxeoDriveActions.java

```diff
@@ -89,6 +104,70 @@ public void testHasOneDriveToken() throws UnsupportedEncodingException {
         revokeToken(token);
     }
 
+    @Test
+    public void testGetDriveEditURL() {
+        /*
+        In the code we are testing, we have the following instructions to retrieve the current server URL:
+        ```
+        ServletRequest servletRequest = (ServletRequest) FacesContext.getCurrentInstance()
+                                                                     .getExternalContext()
+                                                                     .getRequest();
+        String baseURL = VirtualHostHelper.getBaseURL(servletRequest).replaceFirst("://", "/");
+        ```
+        While testing, there is no relevant context, which results in a NullPointerException.
+        To prevent this, we are mocking the objects so that the execution of this snippet results
+        in having the correct baseURL.
+        */
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getServerName()).thenReturn("localhost");
+        when(request.getServerPort()).thenReturn(8080);
+        when(request.getScheme()).thenReturn("http");
+
+        MockExternalContext externalContext = new MockExternalContext();
+        externalContext.setRequest(request);
+        MockFacesContext facesContext = new MockFacesContext(externalContext, new MockApplication());
+        facesContext.setCurrent();
+
+        session.createDocument(session.createDocumentModel("/", "foo", "Folder"));
+        DocumentModel doc = session.createDocumentModel("/foo", "bar", "File");
+
+        Blob mainBlob = new StringBlob("bla bla bla");
+        mainBlob.setFilename("bla.odt");
+        Blob secondBlob = new StringBlob("bli bli bli");
+        secondBlob.setFilename("bli.odt");
+
+        doc.setPropertyValue("file:content", (Serializable) mainBlob);
+        doc.setPropertyValue(
+            "files:files", (Serializable) Collections.singletonList(Collections.singletonMap("file", secondBlob)));
+        doc = session.createDocument(doc);
+
+        String mainBlobURL = nuxeoDriveActions.getDriveEditURL(doc);
+        assertNotNull(mainBlobURL);
+        String expectedMainBlobURL = String.format(
+            "nxdrive://edit/http/localhost:8080/nuxeo/user/Administrator/repo/test/nxdocid/%s/filename/bla.odt/downloadUrl/nxfile/test/%s/blobholder:0/bla.odt", doc.getId(), doc.getId());
+        assertEquals(mainBlobURL, expectedMainBlobURL);
```

Comment:
Should be `assertEquals(expectedMainBlobURL, mainBlobURL)` to comply with the signature: `assertEquals(Object expected, Object actual)`.
This is useful for the assertion failure message: "Expected<...> but was:<...>".